### PR TITLE
DDCE-793: Added Authorised Official Previous Address Query page (PEND…

### PR DESCRIPTION
**Pull request by robert-webb-hmrc**
…ING PRECEDING PAGES REBASE)

# DDCE-793

##Bug fix / 

Added Authorised Official Previous Address page. Has a dependancy on address lookup flow so will need to be rebassed against those changes before this is merged.

## PR Suggestions
- Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?
- Have you done a visual check of the changes?
- Is there any still commented or unused code? Lingering printlns?
- Have things been implemented in a complicated way?
- Are the test still over the coverage threshold?
- Does the compiler throw a lot of warnings? 
- Have methods and tests been named clearly?
- Were there any changes in the config? Do changes need to be made in app-config-???

## Checklist PR Raiser
 - [ ]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [ ]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [ ]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [x]  I've squashed my commits - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

## Checklist PR Reviewer
 - [ ]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [ ]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [ ]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [ ]  I've squashed my commits - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

File: app/controllers/authorisedOfficials/AuthorisedOfficialPreviousAddressController.scala [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import pages.sections.Section7Page
+import play.api.data.Form
+import play.api.mvc._
+import repositories.UserAnswerRepository
+import views.html.authorisedOfficials.AuthorisedOfficialPreviousAddressView
+
+import scala.concurrent.Future
+
+class AuthorisedOfficialPreviousAddressController @Inject()(
+                                                             val sessionRepository: UserAnswerRepository,
```
Comment (abhijeetardale) [inline app/controllers/authorisedOfficials/AuthorisedOfficialPreviousAddressController.scala:36]: alignment please

File: app/controllers/authorisedOfficials/AuthorisedOfficialPreviousAddressController.scala [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    val form: Form[Boolean] = formProvider()
+
+    def onPageLoad(mode: Mode, index: Index): Action[AnyContent] = (identify andThen getData andThen requireData).async { implicit request =>
+
+      getAuthorisedOfficialName(index) {
+        val preparedForm = request.userAnswers.get(AuthorisedOfficialPreviousAddressPage(index)) match {
+          case None => form
+          case Some(value) => form.fill(value)
+        }
+        authorisedOfficialsName =>
```
Comment (abhijeetardale) [inline app/controllers/authorisedOfficials/AuthorisedOfficialPreviousAddressController.scala:54]: please move this to line 50

File: app/navigation/AuthorisedOfficialsNavigator.scala [hunk 3]
```diff
@@ -60,6 +60,12 @@ class AuthorisedOfficialsNavigator @Inject()(implicit frontendAppConfig: Fronten
       case _ =>  routes.SessionExpiredController.onPageLoad()
     }
 
+    case AuthorisedOfficialPreviousAddressPage(index) => userAnswers:UserAnswers  => userAnswers.get(AuthorisedOfficialPreviousAddressPage(index)) match {
+      case Some(true) => routes.DeadEndController.onPageLoad() // TODO redirect to next page once created
```
Comment (abhijeetardale) [inline app/navigation/AuthorisedOfficialsNavigator.scala:64]: can we combine this please Some(_)

File: app/navigation/AuthorisedOfficialsNavigator.scala [hunk 4]
```diff
@@ -95,6 +101,12 @@ class AuthorisedOfficialsNavigator @Inject()(implicit frontendAppConfig: Fronten
...
```
Comment (abhijeetardale) [inline app/navigation/AuthorisedOfficialsNavigator.scala:105]: same as above

File: conf/officialsAndNominees.routes [hunk 5]
```diff
@@ -31,3 +31,7 @@ POST       /:index/authorised-official-enter-national-insurance
 GET        /:index/authorised-official-enter-national-insurance/change          controllers.authorisedOfficials.AuthorisedOfficialsNINOController.onPageLoad(mode: Mode = CheckMode, index: Index)
 POST       /:index/authorised-official-enter-national-insurance/change          controllers.authorisedOfficials.AuthorisedOfficialsNINOController.onSubmit(mode: Mode = CheckMode, index: Index)
 
+GET        /:index/authorised-official-changed-address               controllers.authorisedOfficials.AuthorisedOfficialPreviousAddressController.onPageLoad(mode: Mode = NormalMode, index: Index)
```
Comment (abhijeetardale) [inline conf/officialsAndNominees.routes:34]: please align it.

File: app/navigation/AuthorisedOfficialsNavigator.scala (hunk 3 above)
Comment (robert-webb-hmrc) [reply app/navigation/AuthorisedOfficialsNavigator.scala:64]: Would it make sense to keep the true/false logic for the later pages in the flow?

**Pull request closed without merging**